fine_tuning_mistral_beavertails/beavertails.ipynb

```diff
@@ -0,0 +1,5831 @@
+        "    num_train_epochs=1,\n",
+        "    per_device_train_batch_size=64,\n",
+        "    gradient_accumulation_steps=2,\n",
+        "    gradient_checkpointing=True,\n",
+        "    optim=\"paged_adamw_32bit\",\n",
+        "    logging_steps=50,\n",
+        "    learning_rate=2e-4,\n",
+        "    bf16=False,\n",
+        "    fp16=False,\n",
+        "    max_grad_norm=0.3,\n",
+        "    warmup_ratio=0.03,\n",
+        "    lr_scheduler_type=\"constant\",\n",
+        "    report_to=\"none\",\n",
+        ")\n",
+        "\n",
+        "max_seq_length = 256  # max sequence length for model and packing of the dataset"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 49,
+          "referenced_widgets": [
+            "3ef3d14482a14193bc67018704957cdf",
+            "bc21c240f6244cf29f6faee9c1b11681",
+            "a5f461dade754b4f983a76c21e009b94",
+            "cc9072d85f15418699e5068d60dc0abd",
+            "d122e174a20e4e18ad598609f9c4c1d7",
+            "8016a24045e44812bca6d4b17fb24ad6",
+            "a5a1ecd160cf488ba90b8c5bbff70f5d",
+            "cfdc109d5e474c5680603657e16c7a73",
+            "8d8d72121e20400eb30f29e492ddcc92",
+            "ac1a67f0fc66421cba7c82579e87f015",
+            "52c3e09b9d254a9bb22cfe88907a28d0"
+          ]
+        },
+        "id": "kXk7Mrafg1SN",
+        "outputId": "5ff6d509-bc3e-425a-ef02-fae468b86d46"
+      },
+      "outputs": [
+        {
+          "data": {
+            "application/vnd.jupyter.widget-view+json": {
+              "model_id": "3ef3d14482a14193bc67018704957cdf",
+              "version_major": 2,
+              "version_minor": 0
+            },
+            "text/plain": [
+              "Map:   0%|          | 0/24467 [00:00<?, ? examples/s]"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "trainer = SFTTrainer(\n",
+        "    model=model,\n",
+        "    train_dataset=train_dataset,\n",
+        "    peft_config=peft_config,\n",
+        "    max_seq_length=max_seq_length,\n",
+        "    tokenizer=tokenizer,\n",
+        "    packing=False,\n",
+        "    args=training_args,\n",
+        "    dataset_text_field=\"instructions\",\n",
+        ")\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "LarKF0FIuHI_"
+      },
+      "outputs": [],
+      "source": [
+        "trainer_stats = trainer.train()\n",
+        "train_loss = trainer_stats.training_loss\n",
+        "print(f\"Training loss:{train_loss}\")\n",
+        "\n",
+        "peft_model_id = \"out/\"\n",
+        "trainer.model.save_pretrained(peft_model_id)\n",
+        "tokenizer.save_pretrained(peft_model_id)"
+      ]
+    },
```

Comment:
After this you don't want to show how to generate some text from the fine-tuned model? 

Kind of strange to omit that critical part